src/Snapshot.php

```diff
@@ -90,10 +92,22 @@ protected function shouldIgnoreLine(string $line): bool
 
     protected function loadStream(string $connectionName = null)
     {
+        $directory = (new TemporaryDirectory(config('db-snapshots.temporary_directory_path')))->create();
+
+        config([
+            'filesystems.disks.' . self::class => [
+                'driver' => 'local',
+                'root' => $directory->path(),
+                'throw' => false,
+            ]
+        ]);
+
         LazyCollection::make(function () {
+            Storage::disk(self::class)->writeStream($this->fileName, $this->disk->readStream($this->fileName));
```

Comment:
Can we use `$this->disk` here instead of the facade?

Reply:
@freekmurze i changed the code please check also